src/components/ParametersForm.jsx

```diff
@@ -1,63 +1,96 @@
-import {
-  Button,
-  Center,
-  FormControl, 
-  FormLabel, 
-  NumberInput, 
-  NumberInputField, 
-  NumberInputStepper, 
-  NumberIncrementStepper, 
-  NumberDecrementStepper,
-  Select
-} from '@chakra-ui/react';
 
+import { useState, useEffect } from "react";
+import { api } from "../api";
+import { Button, Center, FormControl, FormLabel, Select,
+  NumberInput, NumberInputField, NumberInputStepper, NumberIncrementStepper, NumberDecrementStepper } from "@chakra-ui/react";
+
+export function ParametersForm({ handleSetJoke }) {
+
+  const [categories, setCategories] = useState([])
+
+  const getJoke = () => {
+    api.get("/random")
+    .then(response => handleSetJoke(response.data.value))
+    .catch(error => console.log(error))
+  }
```

Comment:
This fetch should be in the parent component. So in the props you can pass for example "handleGenerate" function and the logic itself will be in the Page component.

This fetch shouldn't be here. Much better is to have function for fetching in the Page component and this function pass via props (for example prop "handleGenerate" or "onGenerate")